pkg/apis/serving/v1alpha2/kfservice_validation.go

```diff
@@ -64,12 +64,14 @@ func validateKFService(kfsvc *KFService) error {
 	if kfsvc == nil {
 		return fmt.Errorf("Unable to validate, KFService is nil")
 	}
-	if err := validateModelSpec(&kfsvc.Spec.Default); err != nil {
+	if err := validateModelSpec(&kfsvc.Spec.Default.Predict); err != nil {
```

Comment:
change to `validateComponentSpec` instead? Gives other components the right place to put their validate calls...